src/modules/fancyzones/FancyZonesLib/WindowUtils.cpp

```diff
@@ -486,6 +476,58 @@ RECT FancyZonesWindowUtils::AdjustRectForSizeWindowToRect(HWND window, RECT rect
     return newWindowRect;
 }
 
+void FancyZonesWindowUtils::DisableRoundCorners(HWND window) noexcept
+{
+    HANDLE handle = GetPropW(window, ZonedWindowProperties::PropertyCornerPreference);
+    if (!handle)
+    {
+        int cornerPreference = DWMWCP_DEFAULT;
+        // save corner preference if it wasn't set already
+        DwmGetWindowAttribute(window, DWMWA_WINDOW_CORNER_PREFERENCE, &cornerPreference, sizeof(cornerPreference));
+
+        HANDLE preferenceHandle;
+        memcpy(&preferenceHandle, &cornerPreference, sizeof(int));
+
+        if (!SetProp(window, ZonedWindowProperties::PropertyCornerPreference, preferenceHandle))
+        {
+            Logger::error(L"Failed to save corner preference, {}", get_last_error_or_default(GetLastError()));
+        }
+    }
+
+    // Set window corner preference on Windows 11 to "Do not round"
+    int cornerPreference = DWMWCP_DONOTROUND;
+    if (!SUCCEEDED(DwmSetWindowAttribute(window, DWMWA_WINDOW_CORNER_PREFERENCE, &cornerPreference, sizeof(cornerPreference))))
+    {
+        Logger::error(L"Failed to set DWMWCP_DONOTROUND corner preference");
+    }
+}
+
+void FancyZonesWindowUtils::ResetRoundCornersPreference(HWND window) noexcept
+{
+    HANDLE handle = GetPropW(window, ZonedWindowProperties::PropertyCornerPreference);
+    if (handle)
+    {
+        int cornerPreference;
+        memcpy(&cornerPreference, &handle, sizeof(int));
+
+        if (!SUCCEEDED(DwmSetWindowAttribute(window, DWMWA_WINDOW_CORNER_PREFERENCE, &cornerPreference, sizeof(cornerPreference))))
+        {
+            Logger::error(L"Failed to set saved corner preference");
+        }
+
+        RemoveProp(window, ZonedWindowProperties::PropertyCornerPreference);
+    }
+    else
+    {
+        // Set window corner preference on Windows 11 to "Default"
+        int cornerPreference = DWMWCP_DEFAULT;
+        if (!SUCCEEDED(DwmSetWindowAttribute(window, DWMWA_WINDOW_CORNER_PREFERENCE, &cornerPreference, sizeof(cornerPreference))))
```

Comment:
Why are we always setting the DWMWA_WINDOW_CORNER_PREFERENCE to default if we didn't find the previous preference. This means we might overwrite the value if it was different originally.

Reply:
You're right, I'll remove it. I was thinking about previously affected windows.